### PR TITLE
Add event handlers for advancements in slack bot to show in slack chat

### PR DESCRIPTION
**Pull request by polypixeldev**
Advancements, both normal and custom, will now be announced in slack chat when they are completed.

### CODE REVIEW

**Review by the456gamer — COMMENTED** [review]
What about goals, challenges

**Comment by polypixeldev** [comment]
I don't think I specifically tested goals and challenges, but it should work, the only difference is the display frame type, right?

File: src/main/java/com/hackclub/hccore/slack/SlackBot.java [hunk 1]
```diff
@@ -209,6 +217,78 @@ public void onAfkChange(PlayerAFKStatusChangeEvent e) throws IOException {
...
+          advancementType = "challenge";
+      default ->
+          advancementType = "advancement";
+    }
+
+    sendMessage(
+        "%s has completed the %s *%s*".formatted(data.getUsableName(), advancementType, advancementName),
+        playerAdvancementAvatarUrl, "Advancement");
+  }
+
```
Comment (the456gamer) [inline src/main/java/com/hackclub/hccore/slack/SlackBot.java:250]: can this repeated code be refactored? maybe not needed as its only twice.
does not block merging, only if you feel its needed

Comment (polypixeldev) [reply src/main/java/com/hackclub/hccore/slack/SlackBot.java:250]: i think it doesn't really need to be refactored, i don't see a big benefit to doing so.

**Pull request merged**